--- a/newlib/libm/common/s_infconst.c
+++ b/newlib/libm/common/s_infconst.c
@@ -17,5 +17,21 @@
 
 /* Long double version of infinity.  */
 #if defined(_HAVE_LONG_DOUBLE)
-const union __ldmath __infinityld[1] = { { LDBL_MAX+LDBL_MAX } };
+#ifdef __IEEE_BIG_ENDIAN
+ #if LDBL_MANT_DIG == 24
+const union __ldmath __infinityld[1] = {{{0x7f800000, 0, 0, 0}}};
+ #elif LDBL_MANT_DIG == 53
+const union __ldmath __infinityld[1] = {{{0x7ff00000, 0, 0, 0}}};
+ #else
+const union __ldmath __infinityld[1] = {{{0x7fff0000, 0, 0, 0}}};
+#endif /* LDBL_MANT_DIG size  */
+#else /* __IEEE_LITTLE_ENDIAN  */
+ #if LDBL_MANT_DIG == 24
+const union __ldmath __infinityld[1] = {{{0x7f800000, 0, 0, 0}}};
+ #elif LDBL_MANT_DIG == 53
+const union __ldmath __infinityld[1] = {{{0, 0x7ff00000, 0, 0}}};
+ #else
+const union __ldmath __infinityld[1] = {{{0, 0x80000000, 0x00007fff, 0}}};
+#endif /* LDBL_MANT_DIG size  */
+#endif /* __IEEE_LITTLE_ENDIAN  */
 #endif
